File: 1666-make-the-string-great/make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stk = []


        for c in s :
            if stk and ((ord(stk[-1])+32 ==ord(c)) or (ord(stk[-1])-32 ==ord(c))):
                stk.pop(-1)
            else:
                stk.append(c)

        return "".join(stk)

File: 1666-make-the-string-great/make-the-string-great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    const stack = [];
        
        for (const c of s) {
            if (stack.length && (stack[stack.length - 1].charCodeAt(0) + 32 === c.charCodeAt(0) || stack[stack.length - 1].charCodeAt(0) - 32 === c.charCodeAt(0))) {
                stack.pop();
            } else {
                stack.push(c);
            }
        }
        
        return stack.join('');
};